### ISSUE TITLE
Building setup is outdated. package.json has issues. 

### ISSUE THREAD

**Issue by timofei-iatsenko** [MartinCerny-awin/pseudolocale#17]
Hello, we use this package in [`@lingui` ](https://lingui.dev/) project and i would like to help you to make this plugin aligned with actual ecosystem standards. 

1. It's written in typescript but typings don't exposed in npm. 
2. It has bin but it does not stated in package.json
3. test is not executed in CI or ci status is not published
4. looks like content in this repo is not aligned with what is deployed to npm

 if you don't mind then I can do it all

**Comment by MartinCerny-awin** [comment]
Thank you @thekip

**Comment by timofei-iatsenko** [comment]
I have a question related "browser" usage. I see it mentioned in the readme but don't really understand how it supposed to be used.  You should copy file from repository, or what? 

Anyway, i think it's not the case for 2023 and browser usage is already supported via bundler of your choice. What do you think? Does it make sense?

**Comment by timofei-iatsenko** [comment]
@MartinCerny-awin one test is failing in main branch

```

  ● pseudolocale › should pad the string be the specified pad amount

    expect(received).toBe(expected) // Object.is equality

    Expected: 31
    Received: 28

      66 |     const s1 = pseudolocale('this is a test string', options);
      67 | 
    > 68 |     expect(s1.length).toBe(31);
         |                       ^
      69 |   });
      70 | 
      71 |   it('should support a custom start token', function () {

      at Object.<anonymous> (src/test/str.test.ts:68:23)

```

**Comment by timofei-iatsenko** [comment]
It seems the `main` branch has a completely diffrent code than deaployed to npm. I see that current npm version has this usage: 

```js
import pseudolocale from "pseudolocale"

const delimiter = "%&&&%"

pseudolocale.option.delimiter = delimiter
pseudolocale.option.prepend = ""
pseudolocale.option.append = ""

pseudolocale.str(message)
```

Where in the `main` branch it supposed to be used as: 

```ts
import pseudolocale from "pseudolocale"

pseudolocale.str(message, {
 delimiter,
 prepend: "",
 append: ""
})
```

@MartinCerny-awin could you elaborate a bit on that? Did you try to refactor it, but then stuck at some point? Would you want me to finish this version as in the main branch or previous, which is deployed to npm?

**Comment by MartinCerny-awin** [comment]
I would continue with the version which is in GitHub. I am having the same issue with `extend` not working and test failing. I will have a look at that.

**Comment by MartinCerny-awin** [comment]
> I have a question related "browser" usage. I see it mentioned in the readme but don't really understand how it supposed to be used. You should copy file from repository, or what?
> 
> Anyway, i think it's not the case for 2023 and browser usage is already supported via bundler of your choice. What do you think? Does it make sense?

**Comment by MartinCerny-awin** [comment]
@thekip What is difference between these two scripts?

```
    "build": "unbuild",
    "prepublishOnly": "yarn build",
```

**Comment by timofei-iatsenko** [comment]
```
"build": "unbuild",
```
This is standard convention for building. 

```
"prepublishOnly": "yarn build",
``` 

This is made to stop accidental publishing of not-actual `dist`, it is called automatically when you are trying to publish your package.

**Comment by MartinCerny-awin** [comment]
Can we close this issue or is there anything we need to do?

**Comment by timofei-iatsenko** [comment]
I see you published it as v1.3, the changes made to the library were a breaking changes to what was published as 1.2. It should have been published as v2. 

> Can we close this issue or is there anything we need to do?

Regarding concerns raised in this issue everything is done. But i think there is still some room for improvements. Release process + changelog generation could be automated. Semantic versioning should be added.
